File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign=1
        if x<0:
            sign=-1
            x=-x
        rev=0
        while x!=0:
            rev=rev*10+x%10
            x=x//10
        rev=sign*rev
        return  rev if rev in range(-2**31,2**31) else 0

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        
        long rev=0;
        while(x!=0){
            int dig=x%10;
            rev=rev*10+dig;
            x=x/10;
        }

        if(rev>=-Math.pow(2,31) && rev<=Math.pow(2,31)-1){
            return (int)rev;
        }
        return 0;

        
    }
}